crates/sail-plan/src/function/scalar/array.rs

```diff
@@ -12,13 +15,59 @@ fn array_repeat(element: expr::Expr, count: expr::Expr) -> expr::Expr {
     expr_fn::array_repeat(element, count)
 }
 
+fn array_compact(array: expr::Expr) -> expr::Expr {
+    expr_fn::array_remove_all(array, lit(ScalarValue::Null))
+}
+
+fn slice(array: expr::Expr, start: expr::Expr, length: expr::Expr) -> expr::Expr {
+    let start = expr::Expr::Cast(expr::Cast {
+        expr: Box::new(start),
+        data_type: DataType::Int64,
+    });
+    let length = expr::Expr::Cast(expr::Cast {
+        expr: Box::new(length),
+        data_type: DataType::Int64,
+    });
+    let end = expr::Expr::BinaryExpr(BinaryExpr {
+        left: Box::new(start.clone()),
+        op: Operator::Plus,
+        right: Box::new(expr::Expr::BinaryExpr(BinaryExpr {
+            left: Box::new(length),
+            op: Operator::Minus,
+            right: Box::new(lit(ScalarValue::Int64(Some(1)))),
+        })),
```

Comment:
DataFusion mentions that "the to index is exclusive" for `array_slice`. (I'm not sure if the doc is correct though.) Could you take another look? Maybe you can construct some test cases locally and see if they work as expected.